### PR TITLE
#1177 Help reference docs

### PR DESCRIPTION
**Pull request by CarlosNZ**
Implement #1177

Shows a "Help" (external docs) and/or "Reference" menu item depending on whether user is external or internal user ("Admin" permission sees both)).

Requires branch from [back-end PR #781](https://github.com/openmsupply/conforma-server/pull/781)

And here is a template for managing the reference docs:
[manageRefDocs-2.zip](https://github.com/openmsupply/conforma-web-app/files/9230310/manageRefDocs-2.zip) (_Updated with back-end Action changes_)

 
One the template is loaded, you should see "Manage Reference Documents" in the "Admin Configurations" menu.
Screen Shot 2022-07-07 at 3 17 32 PM

Should be fairly straightforward to use. Can add new files, give them descriptions, and change the status of existing reference doc files.

### CODE REVIEW

File: src/containers/User/UserArea.tsx [hunk 1]
```diff
@@ -340,34 +392,53 @@ const getNewDropdownsState = (basepath: string, dropdownsState: DropdownsState):
...
       return {
         dashboard: { active: false },
         templates: { active: true, selection: dropdownsState.templates.selection },
         dataViews: { active: false, selection: '' },
         admin: { active: false, selection: '' },
+        intRefDocs: { active: false, selection: '' },
+        extRefDocs: { active: false, selection: '' },
       }
-    case 'outcomes':
+    case 'data':
```
Comment (CarlosNZ) [inline src/containers/User/UserArea.tsx:407]: Missed this one from a while back

File: src/containers/User/UserArea.tsx [hunk 2]
```diff
@@ -176,6 +194,40 @@ const MainMenuBar: React.FC<MainMenuBarProps> = ({ templates, dataViews }) => {
             />
           </List.Item>
         )}
+        {extReferenceDocs.length && (
```
Comment (andreievg) [inline src/containers/User/UserArea.tsx:197]: not related to changes in this PR but can you please
* add `selectOnBlur={false}` to all of the `<Dropdown` (apart from the reference docs), it's quite annoying that something is selected when menu looses focus

File: src/containers/User/UserArea.tsx [hunk 3]
```diff
@@ -45,20 +51,32 @@ const UserArea: React.FC = () => {
...
+    intReferenceDocs: { uniqueId: string; description: string }[]
+    extReferenceDocs: { uniqueId: string; description: string }[]
+  }
 }
 interface DropdownsState {
   dashboard: { active: boolean }
   templates: { active: boolean; selection: string }
   dataViews: { active: boolean; selection: string }
   admin: { active: boolean; selection: string }
+  intRefDocs: { active: boolean; selection: string }
```
Comment (andreievg) [inline src/containers/User/UserArea.tsx:64]: do we need selection ?

**Review by andreievg — APPROVED** [review]
Looks good, small improvement for something not this PR related.

Also, not related to this PR, I can't really notice `selected-link` class changes, but that's for another day

**Comment by andreievg** [comment]
Should that template have default value set on the file upload ? I can add it if you haven't done it already

File: src/containers/User/UserArea.tsx (hunk 2 above)
Comment (CarlosNZ) [reply src/containers/User/UserArea.tsx:200]: Yeah, good call. ec1e5d4d

File: src/containers/User/UserArea.tsx (hunk 3 above)
Comment (CarlosNZ) [reply src/containers/User/UserArea.tsx]: Yeah, guess not. b1408d7e

**Comment by CarlosNZ** [comment]
> Should that template have default value set on the file upload ? I can add it if you haven't done it already

Default value for what? Not sure why it would it need one.

**Comment by CarlosNZ** [comment]
@andreievg I've posted a new version of the "manage docs" template to handle the new Action structure ("modifyMultiple Records" action)

**Comment by andreievg** [comment]
> Default value for what? Not sure why it would it need one.

Sorry missed the second page

**Comment by andreievg** [comment]
> @andreievg I've posted a new version of the "manage docs" template to handle the new Action structure ("modifyMultiple Records" action)

I've tried the latest one (I enabled the latest template for ref docs), works nicely

**Pull request merged**